src/components/SwapForm/SwapActionButton/SwapOnlyButton.tsx

```diff
@@ -0,0 +1,247 @@
+import useSwapCallbackV3 from 'hooks/useSwapCallbackV3'
+import useTheme from 'hooks/useTheme'
+import { Field } from 'state/swap/actions'
+import { useEncodeSolana } from 'state/swap/hooks'
+import { DetailedRouteSummary } from 'types/route'
+import { checkPriceImpact } from 'utils/prices'
+
+const CustomPrimaryButton = styled(ButtonPrimary).attrs({
+  id: 'swap-button',
+})<{ $minimal?: boolean }>`
+  border: none;
+  font-weight: 500;
+
+  &:disabled {
+    border: none;
+  }
+
+  width: ${({ $minimal }) => ($minimal ? '48%' : '100%')};
+`
+
+export type Props = {
+  minimal?: boolean
+  isAdvancedMode: boolean
+  routeSummary: DetailedRouteSummary | undefined
+  isGettingRoute: boolean
+  isProcessingSwap: boolean
+  isDisabled?: boolean
+
+  currencyIn: Currency | undefined
+  currencyOut: Currency | undefined
+  balanceIn: CurrencyAmount<Currency> | undefined
+  balanceOut: CurrencyAmount<Currency> | undefined
+  parsedAmount: CurrencyAmount<Currency> | undefined
+
+  setProcessingSwap: React.Dispatch<React.SetStateAction<boolean>>
+  setErrorWhileSwap: (e: string) => void
+  buildRoute: () => Promise<BuildRouteResult>
+}
+
+const SwapOnlyButton: React.FC<Props> = ({
+  minimal,
+  isAdvancedMode,
+  routeSummary,
+  isGettingRoute,
+  isProcessingSwap,
+  isDisabled,
+
+  currencyIn,
+  currencyOut,
+  balanceIn,
+  balanceOut,
+  parsedAmount,
+
+  setProcessingSwap,
+  setErrorWhileSwap,
+  buildRoute,
+}) => {
+  const { isSolana } = useActiveWeb3React()
+  const [encodeSolana] = useEncodeSolana()
+  const theme = useTheme()
+  const { mixpanelHandler } = useMixpanel({
+    [Field.INPUT]: currencyIn,
+    [Field.OUTPUT]: currencyOut,
+  })
+  const [buildResult, setBuildResult] = useState<BuildRouteResult>()
+  const [isBuildingRoute, setBuildingRoute] = useState(false)
+  const { priceImpact } = routeSummary || {}
+
+  // the callback to execute the swap
+  const swapCallback = useSwapCallbackV3()
+  const priceImpactResult = checkPriceImpact(priceImpact)
+  const userHasSpecifiedInputOutput = Boolean(currencyIn && currencyOut && parsedAmount)
+  const showLoading = isGettingRoute || isBuildingRoute || ((!balanceIn || !balanceOut) && userHasSpecifiedInputOutput)
+
+  const handleClickSwapForAdvancedMode = async () => {
+    if (!swapCallback || isBuildingRoute) {
+      return
+    }
+
+    setProcessingSwap(true)
+
+    setBuildingRoute(true)
+    const result = await buildRoute()
+    setBuildingRoute(false)
+
+    if (result.error) {
+      setProcessingSwap(false)
+      setErrorWhileSwap(result.error)
+      return
+    }
+
+    if (!result.data?.data || !result.data?.routerAddress) {
+      setProcessingSwap(false)
+      setErrorWhileSwap(t`Build failed. Please try again`)
+      return
+    }
+
+    try {
+      await swapCallback(result.data.routerAddress, result.data.data)
+    } catch (e) {
+      setErrorWhileSwap(e.message)
+    }
+
+    setProcessingSwap(false)
+  }
+
+  const handleClickSwapForNormalMode = async () => {
```

Comment:
do we need try catch

Reply:
add try catch is a must, because when failed it wont run this line `    setBuildingRoute(false)`

Reply:
Understand. `buildRoute` is already wrapped with try/catch